docs/architecture.md

```diff
@@ -82,17 +94,32 @@ def my_command(ctx, pulp_ctx):
     ctx.obj = MyEntityContext(pulp_ctx)
 ```
 
-The named tuple `PluginRequirement` is used to describe dependence on server components.
-It needs the `name` of the plugin and accepts optionally an inclusive `min` version,
-an exclusive `max` version, and an `inverted` flag for exclusion of a version range.
+To declare version restrictions on *options* (*body parameters* in the API), the [`preprocess_entity`][pulp_glue.common.context.PulpEntityContext.preprocess_entity]
+method can check if a given option is being used and conditionally apply the requirements to the context.
+
+Each *plugin* may already have `Contexts` binded to specific [resource_class](/using_the_cli/#general-command-syntax) entities,
+which can be found under `pulp_glue/{plugin-name}/context.py`. In the example, a guard is added to `my_option`, which belongs to the
+`Repository` *resource_class* of `"my_plugin"`.
 
-Additionally, the `PulpOption` provides the `needs_plugins` keyword argument.
-It accepts a list of `PluginRequirements` to error when used.
```

Comment:
Where's this gone?

Reply:
Sorry, I guess I'm in a loophole about this.

When I was asking you about pulp_option vs pulp_glue, I was trying to find out if there was a way of writing:
"use `needs_plugins` at `PulpOption` in this case or at `pulp_glue` in that other case". As version guard can (and generally should) be implemented in `pulp_glue`, I concluded duplicating  `needs_plugin` in `PulpOption` would always be redundant from the CLI perspective, since thats already covered up by pulp_glue.

But if you think it's worth it, I could move this to the "Generics" section, where I mention `pulp_option` (since this section is now a subsection of "Pulp Glue" heading).

Reply:
Oh, so you are saying: `needs_plugins` is redundant now that we try to move that all to `pulp_glue`. You have a point there. But I'm not sure if that is a transition, or whether we actually need that possibility in both places in the end. I cannot come up with a good reason though. Maybe deprecating these is another 1.0 goal. Let's keep it removed and see where we get.